index.js

```diff
@@ -52,13 +56,22 @@ const server = Bun.serve({
       else return new Response('Failed to upgrade.', { status: 500 });
     }
 
-    let pathname = '.' + new URL(req.url).pathname;
-    let file = Bun.file(pathname);
+    let pathname = new URL(req.url).pathname;
+    // URLs with no set pathname will serve /index.html by default
+    if (pathname === '/') {
+      pathname = "/index.html"
+    }
+
+    const filePath = '.' + pathname;
+    const file = Bun.file(filePath);
     if (await file.exists()) {
-      let content = await file.text();
+      let content = await file.bytes();
       if (file.type.includes('text/html')) {
-        let idx = content.search(/<\/body>/i);
-        content = ''.concat(content.slice(0, idx), injection, content.slice(idx));
+        const textDecoder = new TextDecoder();
+        let textContent = textDecoder.decode(content);
+        let idx = textContent.search(/<\/body>/i);
+        textContent = ''.concat(textContent.slice(0, idx), injection, textContent.slice(idx));
+        return new Response(textContent, { headers: { 'Content-Type': file.type } });
```

Comment:
Maybe just `content = textDecoder.decode(content)`, so we don't need an almost identical return statement here.

Reply:
That makes sense. I've made those changes now.